packages/hs-react-ui/src/components/Modal/Modal.tsx

```diff
@@ -97,46 +97,49 @@ const Modal = ({
   closeButtonAttachment = 'inside',
   backgroundBlur = '0.5rem',
   backgroundDarkness = 0.2,
-}: ModalProps) => (
-  <>
-    {closeButtonAttachment === 'corner' && (
-      <StyledCloseButton
-        StyledContainer={StyledCloseButtonContainer}
-        containerProps={{
-          closeButtonAttachment,
-        }}
-        iconPrefix={mdiClose}
-        color={colors.background}
-        elevation={1}
-        type="link"
-        onClick={onClose}
-        {...closeButtonProps}
-      />
-    )}
-    <StyledContainer>
-      {children}
-      {closeButtonAttachment !== 'corner' && (
+}: ModalProps) => {
+  const colors = useColors();
+  return (
+    <>
+      {closeButtonAttachment === 'corner' && (
         <StyledCloseButton
           StyledContainer={StyledCloseButtonContainer}
           containerProps={{
             closeButtonAttachment,
           }}
           iconPrefix={mdiClose}
-          color={closeButtonAttachment === 'inside' ? colors.grayDark : colors.background}
-          elevation={closeButtonAttachment === 'inside' ? 0 : 1}
+          color={colors.background}
+          elevation={1}
           type="link"
```

Comment:
I don't think these changes are right - they should still be in the ternary. The issue is that `type` was changed to `variant` but this file didn't get updated so it is wrong. 
```suggestion
          color={closeButtonAttachment === 'inside' ? colors.grayDark : colors.background}
         elevation={closeButtonAttachment === 'inside' ? 0 : 1}
          variant="text"
```

Reply:
Oh you split it out so that they're different branches